server/src/main/java/org/elasticsearch/index/mapper/vectors/SparseVectorFieldMapper.java

```diff
@@ -115,22 +156,74 @@ public SparseVectorFieldType fieldType() {
     }
 
     @Override
-    public void parse(DocumentParserContext context) {
-        throw new UnsupportedOperationException(ERROR_MESSAGE_7X);
+    protected boolean supportsParsingObject() {
+        return true;
+    }
+
+    @Override
+    public void parse(DocumentParserContext context) throws IOException {
+
+        // No support for indexing / searching 7.x sparse_vector field types
+        if (context.indexSettings().getIndexVersionCreated().before(IndexVersion.V_8_0_0)) {
```

Comment:
Same here - checking that we're not operating on indices created before 8.0. Return the same errors that the previous implementation had.